soc/xtensa/intel_adsp/cavs/gdbstub.c

```diff
@@ -0,0 +1,1394 @@
+	},
+	{
+		/* WINDOWSTART */
+		.idx = 585,
+		.regno = 0x0249,
+		.byte_size = 4,
+		.gpkt_offset = 284,
+		.is_read_only = 1,
+	},
+	{
+		/* PS */
+		.idx = 742,
+		.regno = 0x02E6,
+		.byte_size = 4,
+		.gpkt_offset = 296,
+		.stack_offset = BSA_PS_OFF,
+	},
+	{
+		/* THREADPTR */
+		.idx = 999,
+		.regno = 0x03E7,
+		.byte_size = 4,
+		.gpkt_offset = 300,
+#ifdef CONFIG_THREAD_LOCAL_STORAGE
+		/* Only saved in stack if TLS is enabled */
+		.stack_offset = BSA_THREADPTR_OFF,
+#endif
+	},
+	{
+		/* SCOMPARE1 */
+		.idx = 524,
+		.regno = 0x020C,
+		.byte_size = 4,
+		.gpkt_offset = 308,
+		.stack_offset = BSA_SCOMPARE1_OFF,
+	},
+	{
+		/* EXCCAUSE */
+		.idx = 744,
+		.regno = 0x02E8,
+		.byte_size = 4,
+		.gpkt_offset = 592,
+		.stack_offset = BSA_EXCCAUSE_OFF,
+	},
+	{
+		/* DEBUGCAUSE */
+		.idx = 569,
+		.regno = 0x02E9,
+		.byte_size = 4,
+		.gpkt_offset = 596,
+	},
+	{
+		/* EXCVADDR */
+		.idx = 750,
+		.regno = 0x02EE,
+		.byte_size = 4,
+		.gpkt_offset = 616,
+	},
+	{
+		/* A0 */
+		.idx = 0x0000,
+		.regno = 0x0000,
+		.byte_size = 4,
+		.gpkt_offset = 632,
+		.stack_offset = BSA_A0_OFF,
+	},
+	{
+		/* A1 */
+		.idx = 0x0001,
+		.regno = 0x0001,
+		.byte_size = 4,
+		.gpkt_offset = 636,
+	},
+	{
+		/* A2 */
+		.idx = 0x0002,
+		.regno = 0x0002,
+		.byte_size = 4,
+		.gpkt_offset = 640,
+		.stack_offset = BSA_A2_OFF,
+	},
+	{
+		/* A3 */
+		.idx = 0x0003,
+		.regno = 0x0003,
+		.byte_size = 4,
+		.gpkt_offset = 644,
+		.stack_offset = BSA_A3_OFF,
+	},
+	{
+		/* A4 */
+		.idx = 0x0004,
+		.regno = 0x0004,
+		.byte_size = 4,
+		.gpkt_offset = 648,
+		.stack_offset = -16,
+	},
+	{
+		/* A5 */
+		.idx = 0x0005,
+		.regno = 0x0005,
+		.byte_size = 4,
+		.gpkt_offset = 652,
+		.stack_offset = -12,
+	},
+	{
+		/* A6 */
+		.idx = 0x0006,
+		.regno = 0x0006,
+		.byte_size = 4,
+		.gpkt_offset = 656,
+		.stack_offset = -8,
+	},
+	{
+		/* A7 */
+		.idx = 0x0007,
+		.regno = 0x0007,
+		.byte_size = 4,
+		.gpkt_offset = 660,
+		.stack_offset = -4,
+	},
+	{
+		/* A8 */
+		.idx = 0x0008,
+		.regno = 0x0008,
+		.byte_size = 4,
+		.gpkt_offset = 664,
+		.stack_offset = -32,
+	},
+	{
+		/* A9 */
+		.idx = 0x0009,
+		.regno = 0x0009,
+		.byte_size = 4,
+		.gpkt_offset = 668,
+		.stack_offset = -28,
+	},
+	{
+		/* A10 */
+		.idx = 0x000A,
+		.regno = 0x000A,
+		.byte_size = 4,
+		.gpkt_offset = 672,
+		.stack_offset = -24,
+	},
+	{
+		/* A11 */
+		.idx = 0x000B,
+		.regno = 0x000B,
+		.byte_size = 4,
+		.gpkt_offset = 676,
+		.stack_offset = -20,
+	},
+	{
+		/* A12 */
+		.idx = 0x000C,
+		.regno = 0x000C,
+		.byte_size = 4,
+		.gpkt_offset = 680,
+		.stack_offset = -48,
+	},
+	{
+		/* A13 */
+		.idx = 0x000D,
+		.regno = 0x000D,
+		.byte_size = 4,
+		.gpkt_offset = 684,
+		.stack_offset = -44,
+	},
+	{
+		/* A14 */
+		.idx = 0x000E,
+		.regno = 0x000E,
+		.byte_size = 4,
+		.gpkt_offset = 688,
+		.stack_offset = -40,
+	},
+	{
+		/* A15 */
+		.idx = 0x000F,
+		.regno = 0x000F,
+		.byte_size = 4,
+		.gpkt_offset = 692,
+		.stack_offset = -36,
+	},
+#else
```

Comment:
Maybe add a /* !defined(CONFIG_ADSP_XT_GDB) */ here for those who fell asleep reading the preceding 700 lines of code and don't remember the context? :)